### ISSUE TITLE
With unreadable directory dired-jump shows nothing

### ISSUE THREAD

**Issue by ambihelical** [Fuco1/dired-hacks#99]
When I open dired-jump in a directory with an unreadable directory, the command shows nothing.
If I M-x toggle-debug-on-error before this, the problem seems to be in dired-collapse:

```
Debugger entered--Lisp error: (file-error "Opening directory" "Permission denied" "/home/work/.cache/dconf")
  directory-files("/home/work/.cache/dconf" t)
  f--collect-entries("/home/work/.cache/dconf" nil)
  f-entries("/home/work/.cache/dconf")
  dired-collapse()
  run-hooks(dired-after-readin-hook)
  dired-readin()
  dired-internal-noselect("~/.cache/" nil)
  dired-noselect("~/.cache/" nil)
  #[513 "\300\301\"!\207" [switch-to-buffer dired-noselect] 6 ("/usr/local/share/emacs/25.2/lisp/dired.elc" . 21630) (dired-read-dir-and-switches "")]("~/.cache/" nil)
  ad-Advice-dired(#[513 "\300\301\"!\207" [switch-to-buffer dired-noselect] 6 ("/usr/local/share/emacs/25.2/lisp/dired.elc" . 21630) (dired-read-dir-and-switches "")] "~/.cache/")
  apply(ad-Advice-dired #[513 "\300\301\"!\207" [switch-to-buffer dired-noselect] 6 ("/usr/local/share/emacs/25.2/lisp/dired.elc" . 21630) (dired-read-dir-and-switches "")] "~/.cache/")
  dired("~/.cache/")
  dired-jump(nil nil)
  funcall-interactively(dired-jump nil nil)
  call-interactively(dired-jump nil nil)
  command-execute(dired-jump)

```

**Comment by Fuco1** [comment]
I think this is an error in [f](https://github.com/rejeep/f.el) library used to read the files as it tries to access unreadable directories.

Would you be so kind and open an issue there as well?

**Comment by ambihelical** [comment]
https://github.com/rejeep/f.el/issues/78

**Comment by iquiw** [comment]
Isn't it possible to change to ignore error from `f-entries`, at least until the problem is fixed in `f` ?

``` diff
diff --git a/dired-collapse.el b/dired-collapse.el
index 58e9366..c67eaca 100644
--- a/dired-collapse.el
+++ b/dired-collapse.el
@@ -113,7 +113,7 @@
           (let ((path (dired-get-filename nil t))
                 files)
             (while (and (file-directory-p path)
-                        (setq files (f-entries path))
+                        (setq files (ignore-errors (f-entries path)))
                         (= 1 (length files)))
               (setq path (car files)))
             (setq path (s-chop-prefix (dired-current-directory) path))
```

**Comment by Fuco1** [comment]
Should be fixed via #182.
